data/src/dal/sqlite_dal.rs

```diff
@@ -188,23 +142,63 @@ impl SqliteDal {
         Ok(commands)
     }
 
+    /// Returns the ID of the inserted command
+    pub async fn add_command(
+        &self,
+        command: InternalCommand,
+        tx: Option<&mut Transaction<'_, Sqlite>>,
+    ) -> Result<i64, SqlQueryError> {
+        let current_time = self.get_unix_timestamp()?;
+
+        let query = Query::insert()
+            .into_table(sqlite::Command::Table)
+            .columns([
+                sqlite::Command::Alias,
+                sqlite::Command::Command,
+                sqlite::Command::Tag,
+                sqlite::Command::Note,
+                sqlite::Command::Favourite,
+                sqlite::Command::LastUsed,
+            ])
+            .values_panic([
+                command.alias.into(),
+                command.command.into(),
+                command.tag.into(),
+                command.note.into(),
+                command.favourite.into(),
+                current_time.into(),
+            ])
+            .to_string(SqliteQueryBuilder);
+
+        let result = self
+            .execute_query(&query, tx)
+            .await
+            .map_err(|e| SqlQueryError::InsertCommand(e))?;
```

Comment:
You can do
```suggestion
           .map_err(SqlQueryError::InsertCommand)?;
```
which does the same thing but slightly more clean.